File: problems/0001 - 0400/0202/0202.cpp
class Solution {
public:
    bool isHappy(int n) {

        while (n > 9) {
            int cnt = 0;
            int tmp = n;
            while (tmp) {
                cnt += (tmp % 10) * (tmp % 10);
                tmp /= 10;
            }
            n = cnt;
        }

        if (n == 1 || n == 7) {
            return true;
        }

        return false;

    }
};

File: problems/0001 - 0400/0202/0202.py
class Solution:
    def isHappy(self, n: int) -> bool:
        while n > 9:
            cnt = 0
            temp = n
            while temp:
                cnt += (temp % 10) ** 2
                temp //= 10
            n = cnt

        if n == 1 or n == 7:
            return True
        return False
